### PR TITLE
Add vmap with corresponding tests

### PR DESCRIPTION
**Pull request by mishasinitcyn**
## Description

This PR introduces `xarray_jax.vmap`, a wrapper around `jax.vmap` that supports Xarray data structures by operating on named dimensions.

## Background

This PR adds a wrapper that allows users to specify the mapped dimension by name (e.g., `'batch'`). The implementation follows the logic and style established in the `xarray_jax.pmap` wrapper, as the `jax.vmap` API is based on integer-indexed axes and does not natively work with named dimensions. 

The wrapper forwards `axis_name` and `spmd_axis_name` to`jax.vmap`, providing support for features like parallel collectives (`psum`) and SPMD sharding.

## Changes

- **Added `xarray_jax.vmap` wrapper** in `jax_transforms.py`:
    - Provides a dimension-aware interface to `jax.vmap`
    - Accepts a `dim` string to specify the dimension to map over
    - Forwards `axis_name`, `axis_size`, and `spmd_axis_name` arguments to `jax.vmap`
    - Reuses the established PyTree-handling logic from the `pmap` wrapper
- **Added tests** for `vmap` in `jax_transforms_test.py`:
    - Follows the structure of the `pmap` tests for consistency
    - `test_vmap`: Verifies basic functionality with `xarray_jax.Dataset`
    - `test_vmap_with_jax_coords`: Ensures dynamic `jax_coords` are handled correctly
    - `test_vmap_with_tree_mix_of_xarray_and_jax_array`: Confirms it works on mixed PyTrees containing both Xarray objects and plain JAX arrays
    - `test_vmap_complains_when_dim_not_first`: Checks for correct error handling
    - `test_vmap_axis_name_psum_broadcast`: Validates support for collectives
    - `test_vmap_spmd_axis_name_noop`: Confirms sharding arguments are correctly passed

### CODE REVIEW

File: xarray_jax/jax_transforms.py [hunk 1]
```diff
@@ -19,6 +19,73 @@
...
+    result = fn(*args)
+    nonlocal output_treedef
+    flat_result, output_treedef = jax.tree_util.tree_flatten(result)
+    return flat_result
+
+  vmapped_fn = jax.vmap(
+      fn_passed_to_vmap,
+      in_axes=0,
+      out_axes=0,
+      axis_name=axis_name,
```
Comment (mjwillson) [inline xarray_jax/jax_transforms.py:74]: Good to default the axis name to `dim` if not specified, like we do for pmap.

File: xarray_jax/jax_transforms.py [hunk 2]
```diff
@@ -19,6 +19,73 @@
...
+  """Wraps jax.vmap to handle xarray input/output.
+  This is a convenience wrapper around `jax.vmap` that uses a dimension name
+  instead of integer axes.
+  Constraints:
+    * Any Dataset or DataArray passed to the function must have `dim` as the
+      first dimension. This will be checked. You can ensure this if necessary
+      by calling `.transpose(dim, ...)` beforehand.
+    * All args and return values will be mapped over the first dimension,
+      it will use in_axes=0, out_axes=0. No broadcasting is supported.
+  Args:
```
Comment (mjwillson) [inline xarray_jax/jax_transforms.py:36]: Please stick to the same whitespace conventions used in existing docstrings

**Review by mjwillson — CHANGES_REQUESTED** [review]
Thanks!

Note xarray-based code generally just works in the presence of additional dimensions as xarray handles the broadcasting for you. So something like this might only be needed in fairly niche use cases I think, although still good to have.
